<sect1 id="zend.exception">
    <title>Zend::exception()</title>

    <para>
        Выполняет отложенную загрузку (lazy loading) класса исключения,
        инстанцирует исключение и возвращает его.  
<!--
        Lazy loads an exception class, instantiates the exception, and returns it.
-->
    </para>

    <note>
        <title>Не рекомендуется к использованию<!--Deprecated:--></title>
        <para>
            Этот метод был добавлен в версии 0.6.0. В последнее время было
            решено, что лучше использовать традиционные методы инстанцирования,
            возбуждения и отлова исключений. Метод
            <code>Zend::exception()</code> не рекомендуется к использованию в
            версии 0.6.1 и будет удален из Zend Framework в последующих версиях.
<!--
            This method was introduced in Zend Framework 0.6.0.
            It was recently decided that it's better to use traditional methods
            to instantiate, throw, and catch exceptions.
            The <code>Zend::exception()</code> method is deprecated in
            Zend Framework 0.6.1, and it will be removed from the Zend Framework
            product in a later Preview Release.
-->
        </para>
    </note>

    <programlisting role="php"><![CDATA[<?php

Zend::exception($class, $message, $code);

?>]]></programlisting>

    <para>
        Аргумент <code>$class</code> определяет класс исключения для загрузки и
        инстанцирования. Класс загружается с помощью
        <xref linkend="zend.loadclass" />, поэтому он
        должен следовать соглашению по именованию классов фреймворка и быть
        размещенным по пути, указанному в <code>include_path</code>. 
<!--
        The <code>$class</code> argument specifies the exception class to load
        and instantiate, and is loaded using 
        <xref linkend="zend.loadclass">Zend::loadClass()</xref>; as such, it
        must follow the framework class naming conventions and be somewhere in
        the PHP <code>include_path</code>.
-->
    </para>

    <para>
        Параметры <code>$message</code> и <code>$code</code> являются
        необязательными и передаются в качестве аргументов конструктору
        класса исключения.
<!--
        The <code>$message</code> and <code>$code</code> parameters are both
        optional and passed as arguments to the exception class constructor.
-->
    </para>

    <para>
        В случае успеха производится загрузка класса исключения, создание
        его экземпляра с использованием переданных параметров <code>$message</code> и 
        <code>$code</code>, возвращение созданного объекта. Если
        загрузка класса невозможна или этот класс не наследует от
        класса <code>Exception</code>, то генерируется исключение.
<!--
        If successful, it loads the exception class and creates an instance
        using the <code>$message</code> and <code>$code</code> provided,
        returning the newly created exception instance. If unable to load the
        class or if the class does not extend <code>Exception</code>, it throws
        an exception.
-->
    </para>
</sect1>
<!--
vim:se ts=4 sw=4 et:
-->
